packages/api/src/external/carequality/document/process-outbound-document-retrieval-resps.ts

```diff
@@ -0,0 +1,206 @@
+        errors: issuesWithGateway,
+      },
+      type: "download",
+      requestId,
+      source: MedicalDataSource.CAREQUALITY,
+    });
+
+    await setDocQueryProgress({
+      patient: { id: patientId, cxId: cxId },
+      downloadProgress: {
+        status: "completed",
+      },
+      requestId,
+      source: MedicalDataSource.CAREQUALITY,
+    });
+  } catch (error) {
+    const msg = `Failed to process documents in Carequality.`;
+    console.log(`${msg}. Error: ${errorToString(error)}`);
+
+    await setDocQueryProgress({
+      patient: { id: patientId, cxId: cxId },
+      downloadProgress: { status: "failed" },
+      requestId,
+      source: MedicalDataSource.CAREQUALITY,
+    });
+
+    capture.message(msg, {
+      extra: {
+        context: `cq.processingDocuments`,
+        error,
+        patientId: patientId,
+        requestId,
+        cxId,
+      },
+      level: "error",
+    });
+    throw error;
+  }
+}
+
+async function handleDocReferences(
+  docRefs: DocumentReference[],
+  requestId: string,
+  patientId: string,
+  cxId: string
+) {
+  let errorCountConvertible = 0;
+
+  const { log } = out(`CQ handleDocReferences - requestId ${requestId}, M patient ${patientId}`);
+
+  const currentFHIRDocRefs = await getDocumentsFromFHIR({
+    cxId,
+    patientId,
+    documentIds: docRefs.map(doc => doc.metriportId ?? ""),
+  });
+
+  const transactionBundle: Bundle = {
+    resourceType: "Bundle",
+    type: "transaction",
+    entry: [],
+  };
+
+  for (const docRef of docRefs) {
+    const isDocConvertible = isConvertible(docRef.contentType || undefined);
+    const shouldConvert = isDocConvertible && docRef.isNew;
+
+    if (!docRef.isNew) {
+      errorCountConvertible++;
+    }
+
+    if (shouldConvert) {
+      try {
+        if (!docRef.fileLocation || !docRef.fileName) {
+          throw new Error(`File location or file name is missing for doc ${docRef.metriportId}`);
+        }
+
+        await convertCDAToFHIR({
+          patient: {
+            id: patientId,
+            cxId,
+          },
+          document: {
+            id: docRef.metriportId ?? "",
+            content: { mimeType: docRef.contentType ?? "" },
+          },
+          s3FileName: docRef.fileName,
+          s3BucketName: docRef.fileLocation,
+          requestId,
+          source: MedicalDataSource.CAREQUALITY,
+        });
+      } catch (err) {
+        // don't fail/throw or send to Sentry here, we already did that on the convertCDAToFHIR function
+        log(
+          `Error triggering conversion of doc ${docRef.metriportId}, just increasing errorCountConvertible - ${err}`
+        );
+        errorCountConvertible++;
+      }
+    }
+    const currentFHIRDocRef = currentFHIRDocRefs.filter(
+      fhirDocRef => fhirDocRef.id === docRef.metriportId
+    );
+
+    const docId = docRef.metriportId ?? "";
+
+    const fhirDocRef = cqToFHIR(docId, docRef, patientId, true, currentFHIRDocRef[0]);
```

Comment:
Can we make `currentFHIRDocRef` be a single object, please? Its confusing to read the code and see both `currentFHIRDocRefs` and `currentFHIRDocRef` be an array.